internal/presenters/components.go

```
@@ -0,0 +1,166 @@
+  Ignored issues: {{ .IgnoredIssueCountWithSeverities }} 
+  Open issues:    {{ .OpenIssueCountWithSeverities }}{{ end }}`))
+
+	totalIssueCount := 0
+	openIssueCount := 0
+	ignoredIssueCount := 0
+	openIssueLabelledCount := ""
+	ignoredIssueLabelledCount := ""
+
+	slices.Reverse(summary.SeverityOrderAsc)
+
+	for _, severity := range summary.SeverityOrderAsc {
+		for _, result := range summary.Results {
+			if result.Severity == severity {
+				totalIssueCount += result.Total
+				openIssueCount += result.Open
+				ignoredIssueCount += result.Ignored
+				openIssueLabelledCount += renderInSeverityColor(severity, fmt.Sprintf(" %d %s ", result.Open, strings.ToUpper(severity)))
+				ignoredIssueLabelledCount += renderInSeverityColor(severity, fmt.Sprintf(" %d %s ", result.Ignored, strings.ToUpper(severity)))
+			}
+		}
+	}
+
+	openIssueCountWithSeverities := fmt.Sprintf("%s [%s]", renderBold(strconv.Itoa(openIssueCount)), openIssueLabelledCount)
+	ignoredIssueCountWithSeverities := fmt.Sprintf("%s [%s]", renderBold(strconv.Itoa(ignoredIssueCount)), ignoredIssueLabelledCount)
+	testType := summary.Type
+	if testType == "sast" {
+		testType = "Static code analysis"
+	}
+
+	err := summaryTemplate.Execute(&buff, struct {
+		Org                             string
+		TestPath                        string
+		Type                            string
+		TotalIssueCount                 int
+		IgnoreIssueCount                int
+		OpenIssueCountWithSeverities    string
+		IgnoredIssueCountWithSeverities string
+	}{
+		Org:                             orgName,
+		TestPath:                        testPath,
+		Type:                            testType,
+		TotalIssueCount:                 totalIssueCount,
+		IgnoreIssueCount:                ignoredIssueCount,
+		OpenIssueCountWithSeverities:    openIssueCountWithSeverities,
+		IgnoredIssueCountWithSeverities: ignoredIssueCountWithSeverities,
+	})
+	if err != nil {
+		return fmt.Sprintf("failed to execute summary template: %v", err)
```

Issue: We shouldn't just print errors to stdout, let's return an error object in addition to the string.